src/components/BookList/BookList.js

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { APIS } from '../../config';
+import * as s from '../../pages/ProductDetail/ProductDetailStyled';
+
+const BookList = ({ title, bookList }) => {
+  const [checkedList, setCheckedList] = useState([]);
+  const [showAllBooks, setShowAllBooks] = useState(false);
+
+  const visibleBooks = showAllBooks ? bookList : bookList.slice(0, 7);
+
+  const handleShowMore = () => {
+    setShowAllBooks(prev => !prev);
+  };
+
+  const navigate = useNavigate();
+
+  const handleSingleCheck = (checked, id) => {
+    if (checked) {
+      setCheckedList(prev => [...prev, id]);
+    } else {
+      setCheckedList(checkedList.filter(el => el !== id));
+    }
+  };
+
+  const handleAllCheck = checked => {
+    if (checked) {
+      const idArr = [];
+      bookList.forEach(el => {
+        idArr.push(el.id);
+      });
+      setCheckedList(idArr);
+    } else {
+      setCheckedList([]);
+    }
+  };
+
```

Comment:
```suggestion
    const errorMsg = {
      "NOT_EXIST_TOKEN" : "로그인이 필요합니다",
      "선택된 도서가 없습니다." : '선택된 도서가 없습니다.',
    }
```

아래 리뷰에 필요한 메세지 관리하는 객체입니다!